### PR TITLE
fix: cors not working when specifying specific domain(s)

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

closes: #127 

## Implementation highlights

Change nginx conf and the inputs in web console to support cors multi domain and wildcard in ingestion server.
The following is the table of UI input, supported cors domains and ingestion stack input parameter(ServerCorsOrigin)

UI Input | Valid domains | Input stack string
-- | -- | --
https://example.com | https://example.com | "https://example\.com“
http://example.com | http://example.com | "http://example\.com“
http://\*example\*.com | http://xxxexamplexxx.com, http://example.com | "http://.\*example.\*\\.com“
https://\*example\*.com | https://xxxexamplexxx.com, https://example.com | "https://.\*example.\*\\.com“
http://example.com,https://abc.com | http://example.com, https://abc.com | "http://example\.com\|https://abc\.com"
http://localhost | http://localhost:xxxx | "http://localhost“
http://127.0.0.1 | http://127.0.0.1:xxxx | "http://127.0.0.1“
|* | Any domain | ".*"

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [x] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/test/api/utils.test.ts [hunk 1]
```diff
@@ -267,6 +268,43 @@ describe('Utils test', () => {
...
+      'https://localhost:8080',
+      'localhost,example.com',
+      'localhost, example.com',
+      'localhost:8080,example.com:80',
+      'http://localhost,https://example.com',
+      'abc1.test.com, abc2.test.com, abc3.test.com',
+      'abc1.test.com,abc2.test.com',
+    ];
+    validValues.forEach(v => expect(validatePattern('CORS origin', CORS_PATTERN, v)).toEqual(true));
+    const invalidValues = [
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/test/api/utils.test.ts:294]: add an invalid case `*,abc.com`

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/test/api/utils.test.ts (hunk 1 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/test/api/utils.test.ts:294]: done

**Review by zxkane — CHANGES_REQUESTED** [review]
Also, correct the hint example in web console.

image

**Comment by zxkane** [comment]
> Also, correct the hint example in web console.
> 
> image

@luorobin-a2z how about updating the description of CORS setting to `Enable web requests from specific domains by specifying them in the provided textbox. By default, the ingestion server's CORS origin is empty to disable CORS requests.`?

**Comment by luorobin-a2z** [comment]
Please move CORS setting to above the "additional settings"
Update CORS description as below:
CORS setting allows requests from specific domains to be sent by browsers. This is critical if you are collecting clickstream data from a web application, please input the allowed domain(s) in the following box.

**Review by zxkane — APPROVED** [review]

**Pull request merged**